src/algorithms/reco/InclusiveKinematicsTruth.cc

```diff
@@ -61,7 +53,7 @@ namespace eicrecon {
     }
     const auto pi_p = pi_coll[0].getMomentum();
     const auto pi_p_mag = edm4hep::utils::magnitude(pi_p);
-    const auto pi_mass = pi_coll[0].getPDG() == 2212 ? m_proton : m_neutron;
+    const auto pi_mass = m_particleSvc.particle(pi_coll[0].getPDG()).mass;
```

Comment:
```suggestion
    static const auto pi_mass = m_particleSvc.particle(pi_coll[0].getPDG()).mass;
```

Reply:
`static` would set `pi_mass` to the same value, regardless of the PDG value. The initial nucleon can be proton or neutron, with different mass, and this can change from event to event in e.g. ion beams.

Note: `pi` stands for `p` of the *initial* nucleon, so this is not the pion mass.

Reply:
Yeah, only for the edge case when input has several different beam configurations. But yeah, good thinking, I haven't even considered.